src/content/docs/features/window-state.mdx

```diff
@@ -3,12 +3,152 @@ title: Window State
-</Stub>
+Save window positions and sizes and restore them when the app is reopened.
+
+## Supported Platforms
+
+- Windows
+- Linux
+- macOS
+
+## Setup
+
+_This plugin requires a Rust version of at least **1.75**_
+
+Install the window-state plugin to get started.
+
+<Tabs>
+  <TabItem label="Automatic">
+
+Use your project's package manager to add the dependency:
+
+{' '}
+
+<CommandTabs
+	npm="npm run tauri add window-state"
+	yarn="yarn run tauri add window-state"
+	pnpm="pnpm tauri add window-state"
+	cargo="cargo tauri add window-state"
+/>
+
+  </TabItem>
+  <TabItem label="Manual">
+
+1. Install the Core plugin by adding the following to your `Cargo.toml` file:
+
+```toml title="src-tauri/Cargo.toml"
+[dependencies]
+tauri-plugin-window-state = "2.0.0-beta"
+# alternatively with Git:
+tauri-plugin-window-state = { git = "https://github.com/tauri-apps/plugins-workspace", branch = "v2" }
+```
+
+2. Install the JavaScript Guest bindings using your preferred JavaScript package manager:
+
+  <CommandTabs
+      npm="npm install @tauri-apps/plugin-window-state"
+      yarn="yarn add @tauri-apps/plugin-window-state"
+      pnpm="pnpm add @tauri-apps/plugin-window-state"
+  />
+  </TabItem>
+</Tabs>
+
+## Usage
+
+First you need to register the core plugin with Tauri:
+
+```rust title="src-tauri/src/main.rs" {3}
+fn main() {
+    tauri::Builder::default()
+        .plugin(tauri_plugin_window_state::Builder::default().build())
+        .run(tauri::generate_context!())
+        .expect("error while running tauri application");
+}
+```
+
+Afterwards all windows will remember their state when the app is being closed and will restore to their previous state on the next launch.
+
+Optionally the window-state plugin can also be access with both JavaScript and Rust.
+
+<Tabs>
+<TabItem label="JavaScript">
+
+```javascript
+import { saveWindowState, StateFlags } from "@tauri-apps/plugin-window-state";
+
+saveWindowState(StateFlags.ALL);
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+You can use the `save_window_state()` method exposed by the `AppHandleExt` trait:
+```rust
+use tauri_plugin_window_state::{AppHandleExt, StateFlags};
+
+// `tauri::AppHandle` now has the following additional method
+app.save_window_state(StateFlags::all()); // will save the state of all open windows to disk
+```
+
+</TabItem>
+</Tabs>
+
+Similarly you can manually restore a windows state from disk through JavaScript and Rust.
+
+<Tabs>
+<TabItem label="JavaScript">
+
+```javascript
+import {
+  restoreStateCurrent,
+  StateFlags,
+} from "@tauri-apps/plugin-window-state";
+
+restoreStateCurrent(StateFlags.ALL);
+```
+
+</TabItem>
+<TabItem label="Rust">
+
+You can call the `restore_state()` method exposed by the `WindowExt` trait:
+
+```rust
+use tauri_plugin_window_state::{WindowExt, StateFlags};
+
+// all `Window` types now have the following additional method
+window.restore_state(StateFlags::all()); // will restore the windows state from disk
+```
+
+
+</TabItem>
+</Tabs>
```

Comment:
I think that we would be abusing of that that component tab and since the guide itself isn't that large I think that having 2 separated sections works just fine.